.github/workflows/rust.yml

```diff
@@ -20,6 +20,10 @@ jobs:
       run: cargo build --verbose
     - name: Run tests
       run: cargo test --verbose
+    - name: Build with rayon feature
+      run: cargo build --features rayon --verbose
+    - name: Run tests with rayon feature
+      run: cargo test --features rayon --verbose
```

Comment:
we might need to invert this to run without default features since rayon will be enabled by default